sharding/syncer/service.go

```diff
@@ -69,22 +88,26 @@ func (s *Syncer) handleCollationBodyRequests(feed *event.Feed) {
 		case <-s.ctx.Done():
 			return
 		case req := <-ch:
-			if req.Data == nil {
-				continue
-			}
-			log.Info(fmt.Sprintf("Received p2p request of type: %T", req))
-			res, err := RespondCollationBody(req, s.client, s.shard)
-			if err != nil {
-				log.Error(fmt.Sprintf("Could not construct response: %v", err))
-				continue
+			if req.Data != nil {
+				log.Info(fmt.Sprintf("Received p2p request of type: %T", req))
+				res, err := RespondCollationBody(req, signer, s.shard)
+				if err != nil {
+					s.errChan <- fmt.Errorf("Could not construct response: %v", err)
+					continue
+				}
+				log.Info(fmt.Sprintf("Responding to p2p request with collation with headerHash: %v", res.HeaderHash.Hex()))
+
+				// Notifies the response sent channel for any other handlers that could run upon
+				// this event occurring (also useful for tests.)
+				s.responseSent <- res
+
+				// Reply to that specific peer only.
+				// TODO: Implement this and see the response from the other end.
```

Comment:
Is this TODO for p2p.Send?
If so, can you remove it? 
p2p will be ready soon and I probably wont find this TODO so it will be leftover or irrelevant